### PR TITLE
disable direct reads for log and manifest and add direct io to tests

### PR DESCRIPTION
**Pull request by lightmark**
Disable direct reads for log and manifest. Direct reads should not affect sequential_file
Also add kDirectIO for option_config_ in db_test_util

### CODE REVIEW

File: env/env_posix.cc [hunk 1]
```diff
@@ -152,20 +152,12 @@ class PosixEnv : public Env {
   virtual Status NewSequentialFile(const std::string& fname,
                                    unique_ptr<SequentialFile>* result,
                                    const EnvOptions& options) override {
-    result->reset();
+    // force not using direct io for sequential file
```
Comment (IslamAbdelRahman) [inline env/env_posix.cc]: Does that mean that we only disable it for POSIX, also suppressing the options that I am passing to the Env sounds to me like something that will confuse future developers.

Maybe it's better to construct the optimized options outside the Env and then pass it.
What do you think ?

Comment (lightmark) [inline env/env_posix.cc]: Sure.

File: tools/db_stress.cc [hunk 2]
```diff
@@ -2352,6 +2360,17 @@ int main(int argc, char** argv) {
   SetUsageMessage(std::string("\nUSAGE:\n") + std::string(argv[0]) +
                   " [OPTIONS]...");
   ParseCommandLineFlags(&argc, &argv, true);
+  rocksdb::SyncPoint::GetInstance()->SetCallBack(
```
Comment (ajkr) [inline tools/db_stress.cc]: I'm still kind of confused why we run stress tests or unit tests for direct I/O on platforms that don't support it. And why does db_stress mask O_DIRECT on all platforms while db_test_util masks it on a subset of platforms?

Comment (lightmark) [inline tools/db_stress.cc]: Actually I have to mask it on the subset.

**Pull request closed without merging**